File: online-judge-solutions/Codeforces/855A.cpp
//https://codeforces.com/problemset/problem/855/A
//Lang		:	GNU C++14
//Time		:	15 ms
//Memory	:	200 KB
#include<iostream>
#include<map>
using namespace std;
int main(){
	int n;
	map<string,int> m;
	cin>>n;
	while(n--){
		string s;
		cin>>s;
		m[s]++;
		cout<<(m[s]==1?"NO":"YES")<<"\n";
	}
	return 0;
}

File: online-judge-solutions/Codeforces/855A.py
#https://codeforces.com/problemset/problem/855/A
#Lang		:	Python 3.8
#Time		:	46 ms
#Memory         :	0 KB
n,m=int(input()),{}
for x in range(n):
    s=input()
    m[s]=m.get(s,0)+1
    print("NO" if m[s]==1 else "YES")
